### DISCUSSION TITLE
Error from token::create_collection while publishing contract using object.

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#440] [Questions]
`{
  "Error": "Simulation failed with status: Execution failed in 0x1::account::create_guid at code offset 4\nExecution failed with message: Failed to borrow global resource from 4420c58eec16b9f1e03254dba2a8b22b5ffc3ec37c3951a5cad4ad855deaa4c5"
}`

**Comment by 0xjunha** [comment]
Could you share some context to reproduce the error?

**Reply by ajaythxkur** [reply]
aptos move create-object-and-publish-package \
    --address-name nft_lending \
    --named-addresses nft_lending=$PUBLISHER_ADDR\
    --profile $PUBLISHER_PROFILE \
    --assume-yes --skip-fetch-latest-git-deps

**Reply by 0xjunha** [reply]
Thanks for sharing the context. Could you please share your Move smart contract and Move.toml file as well? It seems the error comes from the Move code, not from the CLI object deployment feature.

**Reply by ajaythxkur** [reply]
got it, i am getting error from this
`fun create_nft_collection(creator: &signer){
        // This means that the supply of the token will not be tracked.
        let maximum_supply = 0;
        // This variable sets if we want to allow mutation for collection description, uri, and maximum.
        // Here, we are setting all of them to false, which means that we don't allow mutations to any CollectionData fields.
        let mutate_setting = vector<bool>[ false, false, false ];
        token::create_collection(
            creator,
            utf8(COLLECTION_NAME),
            utf8(COLLECTION_DESCRIPTION),
            utf8(COLLECTION_URI),
            maximum_supply,
            mutate_setting,
        );
    }
`

**Reply by 0xjunha** [reply]
This error happens since the `create_collection` function in token v1 assumes that the signer is regular account type. This internally calls `account::create_guid`, which attempt to borrow `Account` resource from the signer account. If you call this function in `init_module` and deploy the module with object deployment, that error happens since the object doesn't hold the `Account` resource.

Try using [create_collection function in token v2](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token-objects/sources/aptos_token.move#L72). This will work with your case.

You can either
- deploy your move code to a regular account (`aptos move publish ...`)
- or use the token v2 functions

fyi:
- token v1 (legacy)
  - [docs](https://aptos.dev/en/build/smart-contracts/aptos-token)
  - [code](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/framework/aptos-token)

- token v2
  - [docs](https://aptos.dev/en/build/smart-contracts/digital-asset)
  - [docs](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/framework/aptos-token-objects)
